packages/eas-cli/src/build/utils/resourceClass.ts

```diff
@@ -0,0 +1,135 @@
+  [ResourceClass.LARGE]: BuildResourceClass.AndroidLarge,
+  [ResourceClass.MEDIUM]: BuildResourceClass.AndroidMedium,
+};
+
+async function resolveIosDefaultRequestedResourceClassAsync(
+  getDynamicProjectConfigAsync: DynamicConfigContextFn,
+  projectDir: string,
+  buildProfile: ProfileData<'build'>
+): Promise<ResourceClass> {
+  const { exp } = await getDynamicProjectConfigAsync({ env: buildProfile.profile.env });
+  const { sdkVersion } = exp;
+  const reactNativeVersion = await getReactNativeVersionAsync(projectDir);
+  if (
+    (sdkVersion && semver.satisfies(sdkVersion, '>=48')) ||
+    (reactNativeVersion && semver.satisfies(reactNativeVersion, '>=0.71.0'))
+  ) {
+    return ResourceClass.M1_MEDIUM;
+  } else {
+    return ResourceClass.DEFAULT;
+  }
+}
+
+function resolveAndroidResourceClass(
+  profileResourceClass?: ResourceClass,
+  resourceClassFlag?: ResourceClass
+): BuildResourceClass {
+  if (
+    resourceClassFlag &&
+    [
+      ResourceClass.M1_EXPERIMENTAL,
+      ResourceClass.M1_MEDIUM,
+      ResourceClass.M1_LARGE,
+      ResourceClass.INTEL_MEDIUM,
+    ].includes(resourceClassFlag)
+  ) {
+    throw new Error(`Resource class ${resourceClassFlag} is only available for iOS builds`);
+  }
+
+  const resourceClass = resourceClassFlag ?? profileResourceClass ?? ResourceClass.DEFAULT;
+
+  return androidResourceClassToBuildResourceClassMapping[
+    resourceClass as Exclude<
+      ResourceClass,
+      | ResourceClass.M1_EXPERIMENTAL
+      | ResourceClass.M1_MEDIUM
+      | ResourceClass.M1_LARGE
+      | ResourceClass.INTEL_MEDIUM
+    >
+  ];
+}
+
+async function resolveIosResourceClassAsync(
+  buildProfile: ProfileData<'build'>,
+  getDynamicProjectConfigAsync: DynamicConfigContextFn,
+  projectDir: string,
+  profileResourceClass?: ResourceClass,
+  resourceClassFlag?: ResourceClass
+): Promise<BuildResourceClass> {
+  const resourceClass =
+    resourceClassFlag ??
+    profileResourceClass ??
+    (await resolveIosDefaultRequestedResourceClassAsync(
+      getDynamicProjectConfigAsync,
+      projectDir,
+      buildProfile
+    ));
+
+  if (resourceClass === ResourceClass.M1_EXPERIMENTAL) {
+    Log.warn(`Resource class ${chalk.bold('m1-experimental')} is deprecated.`);
+  }
+
+  if ([ResourceClass.LARGE, ResourceClass.M1_LARGE].includes(resourceClass)) {
+    Log.warn(
+      `Large resource classes are not available for iOS builds yet. Your build will use the medium resource class.`
+    );
+  }
+
+  return iosResourceClassToBuildResourceClassMapping[resourceClass];
+}
+
+export async function resolveBuildResourceClassAsync(
+  profile: ProfileData<'build'>,
+  projectDir: string,
+  getDynamicProjectConfigAsync: DynamicConfigContextFn,
+  resourceClassFlag?: ResourceClass
+): Promise<BuildResourceClass> {
+  const profileResourceClass = profile.profile.resourceClass;
+  if (profileResourceClass && resourceClassFlag && resourceClassFlag !== profileResourceClass) {
+    Log.warn(
```

Comment:
In response to https://github.com/expo/eas-cli/pull/1655/files#r1087760626

This message suggest that flag is more important option than eas.json. I don't have an opinion on actual behavior and which field should have priority, but in other places where one config can override another, we are doing it silently.

If you want to keep it in add to this message that flag is deprecated.

Reply:
We already print the deprecation warning as the first thing when the user calls `eas build` with the `--resource-class` flag. Do you think it's fine to print the same information twice? Just want to make sure that you are aware of the first warning.

<img width="716" alt="Screenshot 2023-01-26 at 13 31 01" src="https://user-images.githubusercontent.com/55145344/214836141-8dd49a27-1157-4bf6-8c46-07e3b9592917.png">

Reply:
The whole experience looks like this at the moment:

<img width="713" alt="Screenshot 2023-01-26 at 13 32 59" src="https://user-images.githubusercontent.com/55145344/214836518-d25a6c59-907a-4795-a7a3-45cb6482ce72.png">

Reply:
then I think this one is not necessary